www/js/app.js

```diff
@@ -998,16 +1001,47 @@ define(['jquery', 'zimArchiveLoader', 'util', 'uiUtil', 'cookies','abstractFiles
             document.getElementById('articleContent').contentWindow.focus();
             $("#searchingArticles").show();
             if (dirEntry.isRedirect()) {
-                selectedArchive.resolveRedirect(dirEntry, readArticle);
+                // Update expectedArticleURLToBeDisplayed before redirect. 
+                selectedArchive.resolveRedirect(dirEntry, function (resolvedDirEntry) {
+                    updatedExpectedURLOnRedirect(resolvedDirEntry, readArticle);
+                });
             } else {
                 params.isLandingPage = false;
+                expectedArticleURLToBeDisplayed  = dirEntry.namespace + "/" + dirEntry.url;
                 readArticle(dirEntry);
             }
         } else {
             alert("Data files not set");
         }
     }
 
+    /**
+     * Check whether the given URL from given dirEntry equals the expectedArticleURLToBeDisplayed
+     * @param {DirEntry} dirEntry The directory entry of the article to read
+     */
+    function entryIsToExpectedBeDisplayed(dirEntry) {
```

Comment:
The name of this function seems difficult to understand for me.
I would suggest `isDirEntryExpectedToBeDisplayed`